plugin/trino-hive/src/test/java/io/trino/plugin/hive/containers/Hive4Metastore.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive.containers;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.common.net.HostAndPort;
+import io.airlift.log.Logger;
+import io.trino.testing.containers.BaseTestContainer;
+import org.testcontainers.containers.Network;
+
+import java.net.URI;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+public class Hive4Metastore
+        extends BaseTestContainer
+{
+    private static final Logger log = Logger.get(HiveHadoop.class);
+    private static final String HOST_NAME = "metastore";
+    public static final String HIVE4_IMAGE = "ghcr.io/trinodb/testing/hive4.0-hive:latest";
```

Comment:
Add to the description of the PR that it depends on https://github.com/trinodb/docker-images/pull/218